<launch>
  <!-- global PARAMS Change here if needed -->
  <arg name="gui" default="False" />
  <arg name="autostart" default="true"/>
  <arg name="serial_device" default="/dev/ttyUSB0"/>
  <arg name="simulation" default="False"/>
  <arg name="logging_config" default="$(find schunk_svh_driver)/etc/logging.xml" />
  <arg name="use_internal_logging" default="False"/>
  <arg name="controller_config" default="$(find schunk_svh_driver)/etc/controller_user.yaml" />
  <arg name="general_config" default="$(find schunk_svh_driver)/etc/svh.yaml" />
  <arg name="name_prefix" default="left_hand" />
  <arg name="frequency" default="30" />
  <arg name="trajectory_controller_name" />

  <!-- load basic configuration -->
  <rosparam file="$(arg general_config)" command="load" ns="svh_ros_control_node"/>
  <!-- Load Yaml Configuration file-->
  <rosparam file="$(find schunk_svh_driver)/etc/controller_default.yaml" command="load" ns="svh_ros_control_node"/>
  <!-- Load the User Config file which will overwrite values from default -->
  <rosparam file="$(arg controller_config)" command="load" ns="svh_ros_control_node"/>


  <!-- Launch actual driver node (if not in simulation)-->
  <node unless="$(arg simulation)" name="svh_ros_control_node" pkg="schunk_svh_driver" type="svh_ros_control_node" output="screen">
    <param name="serial_device" value="$(arg serial_device)"/>
    <param name="autostart" value="$(arg autostart)" />
    <!-- Params for the ICL (driver internal) logging. General configuration in the given logging.xml, output will be saved to the who given files -->
    <param name="logging_config" value="$(arg logging_config)" />
    <param name="use_internal_logging" value="$(arg use_internal_logging)" />
    <param name="name_prefix" value="$(arg name_prefix)" />
    <param name="frequency"           type="int"      value="$(arg frequency)" />
    <param name="use_ros_control" type="bool" value="True"/>
    <param name="trajectory_controller_name" value="$(arg trajectory_controller_name)" />
  </node>

</launch>
